### PR TITLE
[YANG]: Add Yang model support for adding dom_polling to PORT table

### PR DESCRIPTION
**Pull request by mihirpat1**
#### Why I did it
Added YANG related changes for adding `dom_polling` field in PORT table of CONFIG_DB. This field can be set with `config interface transceiver dom PORT_NAME (enable|disable)` CLI.

The `dom_polling` field was added through https://github.com/sonic-net/sonic-utilities/pull/3187. Please refer to this PR for the details on the reason for adding `dom_polling` field.

##### Work item tracking
- Microsoft ADO **(number only)**:
27069573

#### How I did it
Added `dom_polling` field to CONFIG_DB PORT table.

#### How to verify it
Added unit tests for both valid and invalid options for controlling `dom_polling`.
Valid values for for `dom_polling` are `enabled` and `disabled`
Any other value is treated as an invalid value

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [x] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

**Review by prgeor — DISMISSED** [review]

**Comment by prgeor** [comment]
@qiluo-msft can you review/approve

File: src/sonic-yang-models/yang-models/sonic-port.yang [hunk 1]
```diff
@@ -161,6 +161,13 @@ module sonic-port{
 					}
 				}
 
+				leaf dom_polling {
+
+					type string {
+						pattern "enabled|disabled";
```
Comment (qiluo-msft) [inline src/sonic-yang-models/yang-models/sonic-port.yang:167]: this attribute is not required. What is the default value if missing?

Comment (qiluo-msft) [reply src/sonic-yang-models/yang-models/sonic-port.yang:167]: Is the default value backward compatible?

Comment (mihirpat1) [reply src/sonic-yang-models/yang-models/sonic-port.yang:167]: The default value is `enabled` and is backward compatible.

File: src/sonic-yang-models/yang-models/sonic-port.yang [hunk 2]
```diff
@@ -161,6 +161,13 @@ module sonic-port{
 					}
 				}
 
+				leaf dom_polling {
+
+					type string {
```
Comment (ganglyu) [inline src/sonic-yang-models/yang-models/sonic-port.yang:166]: https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-yang-models/yang-templates/sonic-types.yang.j2#L127-L132
Can you use admin_mode from sonic-types.yang?

Comment (mihirpat1) [reply src/sonic-yang-models/yang-models/sonic-port.yang]: @ganglyu - I have addressed this now.

**Review by ganglyu — APPROVED** [review]
lgtm

**Review by wen587 — APPROVED** [review]

**Review by prgeor — APPROVED** [review]

**Comment by mihirpat1** [comment]
@qiluo-msft Can you please help in merging this PR?

**Comment by StormLiangMS** [comment]
hi @mihirpat1 there is conflict with 202305, could you file separate PR and test with 20203 image?

**Comment by mihirpat1** [comment]
> hi @mihirpat1 there is conflict with 202305, could you file separate PR and test with 20203 image?

Hi @StormLiangMS,
I have now created the below PR for 202305.
https://github.com/sonic-net/sonic-buildimage/pull/18343
MSFT ADO - 27069573

**Comment by mihirpat1** [comment]
@yxieca It will be great if you can help to cherry-pick the below PR for 202311.
https://github.com/sonic-net/sonic-buildimage/pull/18344
MSFT ADO - 27069573

**Pull request merged**